### PR TITLE
refactor(home): refactor home page

### PR DESCRIPTION
**Pull request by Oxiang**
## Problem

Migrating jsx files in home folder to tsx files

## Solution

- change jsx file into tsx file
- switch redux higher component connect to redux hooks
- replace propTypes with typescript

### CODE REVIEW

File: src/client/home/components/StatisticsSliver.tsx
Comment (yong-jie) [inline src/client/home/components/StatisticsSliver.tsx:83]: this sounds like a problem that should have been solved in the redux level, like both in the initial state of the reducer and in the api-call handlers that ensure to never set it to null

does it make more sense to solve this nullability problem in the redux initial state and api-call-result-handler rather than here?

File: src/client/home/components/LandingGraphicSilver.tsx
Comment (yong-jie) [inline src/client/home/components/LandingGraphicSilver.tsx:161]: this invariant is only true right now with our current code. X months later if someone deletes the element, this assertion is going to come and bite us. Is it possible to use optional chaining here? `document.getelementbyid()?.scrollIntoView()` etc

**Comment by Oxiang** [comment]
Will rebase once approved review

File: src/client/home/actions/index.ts
Comment (yong-jie) [inline src/client/home/actions/index.ts:46]: just to check: back then when this check was for null, its purpose was to see if this `loadStats`  function ran before right? if that's the case, this time you have refactored the `useEffect` to only run once so you wont ever need any check here at all

**Review by yong-jie — COMMENTED** [review]
got one more quick question before we merge this

File: src/client/home/actions/index.ts
Comment (Oxiang) [reply src/client/home/actions/index.ts]: make sense, removed the checks!

**Review by yong-jie — APPROVED** [review]
lgtm!

**Review by yong-jie — CHANGES_REQUESTED** [review]
the PR looks good to me, but something looks very wrong with the rebase so i dismissed my review

**Pull request closed without merging**
